tests/tools.vitruv.applications.umljava.tests/src/tools/vitruv/applications/umljava/tests/util/JavaUmlViewFactory.xtend

```diff
@@ -0,0 +1,124 @@
+package tools.vitruv.applications.umljava.tests.util
+
+import org.eclipse.xtend.lib.annotations.FinalFieldsConstructor
+import tools.vitruv.framework.vsum.views.ViewProvider
+import org.eclipse.uml2.uml.Model
+import org.emftext.language.java.containers.CompilationUnit
+import java.util.Collection
+import tools.vitruv.framework.vsum.views.ViewTypeFactory
+import tools.vitruv.framework.vsum.views.View
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.CoreMatchers.not
+import static org.hamcrest.CoreMatchers.equalTo
+import org.emftext.language.java.containers.Package
+
+@FinalFieldsConstructor
+class JavaUmlViewFactory {
+	val ViewProvider viewProvider
+
+	private def View createUmlView() {
+		createViewOfElements("UML", #{Model})
+	}
+
+	private def View createJavaView() {
+		createViewOfElements("Java packages and classes", #{Package, CompilationUnit})
+	}
+
+	private def View createUmlAndJavaClassesView() {
+		createViewOfElements("UML and Java classes", #{CompilationUnit, Model})
+	}
+
+	private def View createUmlAndJavaPackagesView() {
+		createViewOfElements("UML and Java packages", #{Package, Model})
+	}
+
+	private def View createViewOfElements(String viewName, Collection<Class<?>> rootTypes) {
+		val selector = viewProvider.createSelector(ViewTypeFactory.createIdentityMappingViewType(viewName))
+
+		for (rootElement : selector.selectableElements.filter[element|rootTypes.exists[it.isInstance(element)]]) {
+			selector.setSelected(rootElement, true)
+		}
+		val view = selector.createView()
+		assertThat("view must not be null", view, not(equalTo(null)))
+		return view
+	}
+
+	/**
+	 * Changes the given view according to the given modification function, commits the performed changes
+	 * and closes the view afterwards.
+	 */
+	private def void changeView(View view, (View)=>void modelModification) {
+		modelModification.apply(view)
+		view.commitChanges()
+		view.close()
+	}
+
+	/**
+	 * Changes the UML view containing all UML models as root elements according to the given modification 
+	 * function, commits the performed changes and closes the view afterwards.
+	 */
+	def void changeUmlView((View)=>void modelModification) {
+		changeView(createUmlView) [
+			modelModification.apply(it)
+		]
+	}
```

Comment:
This is rather a stylistic question but why the need to create another closure when we can just use the compact form `changeView(createUmlView, modelModification)`.
Applies to multiple following methods too.

Reply:
I had two reasons for that:
1. Compactness and ease of understanding: You can write `changeUmlView []` rather than `changeView(createUmlView) []`, which is a little more compact and a little easier to read, but, of course, does not justify these methods alone.
2. Ensure closing views: The risk when providing a `createUmlView` (and other similiar) method(s) is that one might forget to close the view after using it. This happened to me and resulted in tests using large amounts of memory. To enforce that views get closed, I chose to not provide the methods to create them (those methods are private to the utility class) but only methods that provide a transaction of creating a view, executing something on them and then closing them. 

That's the basic rationale behind the design. Feel free to comment and discuss.

Reply:
I was not questioning the method but its implementation as you can make the **method body** more compact by replacing it with `changeView(createUmlView, modelModification)`.

Reply:
Sorry, I completely misunderstood you. Thank you for clarification. Of course, there is no reason for the complicated implementation of the method body. I've fixed that in d1da6a1.

Reply:
Thanks. I think with the latest changes you accidentally made the `createUmlView` method public. At least I don't see a reason to remove the private modifier.

Reply:
Don't commit after 6pm 😉 Thank you for checking the commit!